nni/tools/nnictl/nnictl_utils.py

```diff
@@ -222,9 +222,18 @@ def stop_experiment(args):
     if experiment_id_list:
         for experiment_id in experiment_id_list:
             print_normal('Stopping experiment %s' % experiment_id)
-            experiments_config = Experiments()
-            experiments_dict = experiments_config.get_all_experiments()
-            rest_pid = experiments_dict.get(experiment_id).get('pid')
+            # retry up to 10 times to get the experiment metadata
+            for i in range(1, 11):
+                experiments_dict = Experiments().get_all_experiments()
+                experiment_info = experiments_dict.get(experiment_id)
```

Comment:
Could it be possible that the experiment doesn't exist at all? Maybe `experiment_id` is wrong.

Reply:
`experiment_id` should be in `.experiemnt`, because we get this `experiment_id` from `.experiemnt`, but yes, maybe `experiment_id` is deleted from this file before we read it, or the whole `.experiemnt` file is deleted...

Reply:
I think this may mainly cause by the file being modified when we read the file... the lock will be expired in 2s, but the action on file is over 2s, I'm not sure.